src/conf_mode/service_monitoring_telegraf.py

```diff
@@ -89,20 +97,21 @@ def verify(monitoring):
 
 def generate(monitoring):
     if not monitoring:
-        # Delete config file
-        if os.path.isfile(config_telegraf):
-            os.unlink(config_telegraf)
-        if os.path.isfile(systemd_override):
-            os.unlink(systemd_override)
-        if os.path.isfile(syslog_telegraf):
-            os.unlink(syslog_telegraf)
+        # Delete config and systemd files
+        config_files = [config_telegraf, systemd_telegraf_service, systemd_override, syslog_telegraf]
+        for file in config_files:
+            if os.path.isfile(file):
+                os.unlink(file)
 
+        # Delete old directories
+        if os.path.isdir(cache_dir):
+            rmtree(cache_dir, ignore_errors=True)
 
         return None
 
     # Create telegraf cache dir
     if not os.path.exists(cache_dir):
-        os.mkdir(cache_dir)
+        os.makedirs(cache_dir)
 
     chown(cache_dir, 'telegraf', 'vyattacfg')
```

Comment:
While this is unrelated to the issue on hand, I feel uneasy about hardcoding `vyattacfg` in new scripts. Can we make it a constant somewhere in `vyos.util`, or a function like `make_vyos_group_writable`?